contracts/AbstractConference.sol

```diff
@@ -324,4 +334,50 @@ contract AbstractConference is Conference, GroupAdmin {
         revert('tokenAddress must be impelmented in the child class');
     }
 
+    /* ERC721 implementation start */
+
+    function tokenURI(uint256 tokenId) public view returns (string memory) { 
+        return string(abi.encodePacked(baseTokenURI, Utils.addr2str(address(this)), '/', _tokenURI(tokenId)));
+    }
+    
+    /**
+     * @dev Internal returns an URI for a given token ID.
+     * Throws if the token ID does not exist. May return an empty string.
+     * @param tokenId uint256 ID of the token to query
+     */
+    function _tokenURI(uint256 tokenId) internal view returns (string memory) { 
+        require(_exists(tokenId), 'ERC721Metadata: URI query for nonexistent token');
+        return _tokenURIs[tokenId];
+    }
+
+    /**
+     * @dev Mints a new token.
+     * @param to The address that will own the minted token
+     * @param tokenId uint256 ID of the token to be minted
+     */
+    function mint(address to, uint256 tokenId) internal {
+        _mint(to, tokenId);
+        _tokenURIs[tokenId] = Utils.uint2str(tokenId);
+    }
+
+    function transferFrom(address from, address to, uint256 tokenId) public onlyActive {
```

Comment:
If you rename `transferFrom` to `_beforeTokenTransfer` you don't need to override `safeTransferFrom(address from, address to, uint256 tokenId)` nor `safeTransferFrom(address from, address to, uint256 tokenId, bytes memory _data)`

Reply:
Oh, is it because we still use the old version of the solidity? Ok, then let's leave it as is.